numpyro/distributions/discrete.py

```diff
@@ -405,6 +405,67 @@ def Categorical(probs=None, logits=None, validate_args=None):
         raise ValueError("One of `probs` or `logits` must be specified.")
 
 
+class DiscreteUniform(Distribution):
+    arg_constraints = {"low": constraints.dependent, "high": constraints.dependent}
+    has_enumerate_support = True
+
+    def __init__(self, low=0, high=1, validate_args=None):
+        self.low, self.high = promote_shapes(low, high)
+        batch_shape = lax.broadcast_shapes(jnp.shape(low), jnp.shape(high))
+        self._support = constraints.integer_interval(low, high)
+        super().__init__(batch_shape, validate_args=validate_args)
+
+    @constraints.dependent_property(is_discrete=True, event_dim=0)
+    def support(self):
+        return self._support
+
+    def sample(self, key, sample_shape=()):
+        shape = sample_shape + self.batch_shape
+        return random.randint(key, shape=shape, minval=self.low, maxval=self.high + 1)
+
+    @validate_sample
+    def log_prob(self, value):
+        shape = lax.broadcast_shapes(jnp.shape(value), self.batch_shape)
+        return -jnp.broadcast_to(jnp.log(self.high + 1 - self.low), shape)
+
+    def cdf(self, value):
+        cdf = (jnp.floor(value) + 1 - self.low) / (self.high - self.low + 1)
+        return jnp.clip(cdf, a_min=0.0, a_max=1.0)
+
+    def icdf(self, value):
+        return self.low + value * (self.high - self.low + 1) - 1
+
+    @property
+    def mean(self):
+        return self.low + (self.high - self.low) / 2.0
+
+    @property
+    def variance(self):
+        return ((self.high - self.low + 1) ** 2 - 1) / 12.0
+
+    def enumerate_support(self, expand=True):
+        if not not_jax_tracer(self.high) or not not_jax_tracer(self.low):
+            raise NotImplementedError("Both `low` and `high` must not be a JAX Tracer.")
+        low = np.amax(self.low)
+        if np.amin(low) != self.low:
```

Comment:
In case low is an array, we need to check `np.any(np.amax(self.low) != self.low)`. Similar for `high`.